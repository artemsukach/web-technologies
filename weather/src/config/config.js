import snow from '../img/snow.png';
import wind from '../img/wind.png';
import sun from '../img/sun.png';
import rain from '../img/rain.png';

export const WEATHER = [
  {
    date: '02/28/2022',
    minTemperature: -3,
    maxTemperature: +4,
    atmosphericPressure: 720,
    relativeHumidity: 80,
    windSpeed: 8,
    windDirection: 'South',
    description: snow,
  },
  {
    date: '03/01/2022',
    minTemperature: -2,
    maxTemperature: +6,
    atmosphericPressure: 735,
    relativeHumidity: 77,
    windSpeed: 3.9,
    windDirection: 'South-East',
    description: snow,
  },
  {
    date: '03/02/2022',
    minTemperature: -2,
    maxTemperature: +5,
    atmosphericPressure: 740,
    relativeHumidity: 63,
    windSpeed: 4.3,
    windDirection: 'North',
    description: wind,
  },
  {
    date: '03/03/2022',
    minTemperature: -1,
    maxTemperature: +3,
    atmosphericPressure: 730,
    relativeHumidity: 73,
    windSpeed: 3.4,
    windDirection: 'North-West',
    description: rain,
  },
  {
    date: '03/04/2022',
    minTemperature: 0,
    maxTemperature: +6,
    atmosphericPressure: 729,
    relativeHumidity: 87,
    windSpeed: 3.0,
    windDirection: 'East',
    description: wind,
  },
  {
    date: '03/05/2022',
    minTemperature: 1,
    maxTemperature: 8,
    atmosphericPressure: 738,
    relativeHumidity: 63,
    windSpeed: 3.3,
    windDirection: 'West',
    description: sun,
  },
  {
    date: '03/06/2022',
    minTemperature: 2,
    maxTemperature: 9,
    atmosphericPressure: 742,
    relativeHumidity: 78,
    windSpeed: 3.4,
    windDirection: 'South-West',
    description: sun,
  },
];

export const DAYS = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday',
];

export const MONTHS = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];
