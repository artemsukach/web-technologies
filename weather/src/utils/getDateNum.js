export const getDateNum = (date) => {
  return date.getDate();
};