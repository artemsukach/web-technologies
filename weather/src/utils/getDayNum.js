export const getDayNum = (date) => {
  return date.getDay();
};