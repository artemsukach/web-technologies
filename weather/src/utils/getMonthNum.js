export const getMonthNum = (date) => {
  return date.getMonth();
};