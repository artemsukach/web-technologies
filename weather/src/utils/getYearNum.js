export const getYearNum = (date) => {
  return date.getFullYear();
};