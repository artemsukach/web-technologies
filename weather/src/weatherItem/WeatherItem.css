.weather__item {
  border-radius: 10px;
  background-color: #e0e0e8;
  text-align: center;
}

.weather__item+.weather__item{
  margin-left: 10px;
}