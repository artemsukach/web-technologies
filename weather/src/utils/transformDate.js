import { MONTHS, DAYS } from '../config/config';

export const transformDate = (
  date,
  getMonthNum,
  getDateNum,
  getDayNum,
  getYearNum
) => {
  const monthNum = getMonthNum(date);
  const dateNum = getDateNum(date);
  const dayNum = getDayNum(date);
  const year = getYearNum(date);

  const month = MONTHS[monthNum];
  const day = DAYS[dayNum];

  return {
    day,
    dateNum,
    month,
    year,
  };
};

