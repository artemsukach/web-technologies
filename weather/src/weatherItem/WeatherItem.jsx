import React from 'react';
import './WeatherItem.css';
import PropTypes from 'prop-types';
import { transformDate } from '../utils/transformDate';
import { getMonthNum } from '../utils/getMonthNum';
import { getDateNum } from '../utils/getDateNum';
import { getDayNum } from '../utils/getDayNum';
import { getYearNum } from '../utils/getYearNum';
import { WeatherDay } from '../weatherItemComponent/WeatherDay.jsx';
import { WeatherDate } from '../weatherItemComponent/WeatherDate.jsx';
import { WeatherMonth } from '../weatherItemComponent/WeatherMonth.jsx';
import { WeatherYear } from '../weatherItemComponent/WeatherYear.jsx';
import { WeatherDescription } from '../weatherItemComponent/WeatherDescription.jsx';
import { WeatherTemperature } from '../weatherItemComponent/WeatherTemperature.jsx';
import { WeatherPressure } from '../weatherItemComponent/WeatherPressure.jsx';
import { WeatherHumidity } from '../weatherItemComponent/WeatherHumidity.jsx';
import { WeatherWindSpeed } from '../weatherItemComponent/WeatherWindSpeed.jsx';
import { WeatherWindDirection } from '../weatherItemComponent/WeatherWindDirection.jsx';

function WeatherItem({
  date,
  minTemperature,
  maxTemperature,
  atmosphericPressure,
  relativeHumidity,
  windSpeed,
  windDirection,
  description,
}) {
  const newDate = new Date(date);
  const { day, dateNum, month, year } = transformDate(
    newDate,
    getMonthNum,
    getDateNum,
    getDayNum,
    getYearNum
  );

  return (
    <div className="weather__item">
      <WeatherDay day={day} />
      <WeatherDate dayNum={dateNum} />
      <WeatherMonth month={month} />
      <WeatherYear year={year} />
      <WeatherDescription description={description} />
      <WeatherTemperature
        minTemperature={minTemperature}
        maxTemperature={maxTemperature}
      />
      <WeatherPressure atmosphericPressure={atmosphericPressure} />
      <WeatherHumidity relativeHumidity={relativeHumidity} />
      <WeatherWindSpeed windSpeed={windSpeed} />
      <WeatherWindDirection windDirection={windDirection} />
    </div>
  );
}

WeatherItem.propTypes = {
  date: PropTypes.string,
  minTemperature: PropTypes.number,
  maxTemperature: PropTypes.number,
  atmosphericPressure: PropTypes.number,
  relativeHumidity: PropTypes.number,
  windSpeed: PropTypes.number,
  windDirection: PropTypes.string,
  description: PropTypes.string,
}

export default WeatherItem;
