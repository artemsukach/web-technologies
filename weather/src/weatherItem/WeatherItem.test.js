import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import WeatherItem from './WeatherItem';
import { WEATHER } from '../config/config';
import { WeatherDay } from '../WeatherItemComponent/WeatherDay.jsx';
import { WeatherDate } from '../WeatherItemComponent/WeatherDate.jsx';
import { WeatherMonth } from '../WeatherItemComponent/WeatherMonth.jsx';
import { WeatherYear } from '../WeatherItemComponent/WeatherYear.jsx';
import { WeatherDescription } from '../WeatherItemComponent/WeatherDescription.jsx';
import { WeatherTemperature } from '../WeatherItemComponent/WeatherTemperature.jsx';
import { WeatherPressure } from '../WeatherItemComponent/WeatherPressure.jsx';
import { WeatherHumidity } from '../WeatherItemComponent/WeatherHumidity.jsx';
import { WeatherWindSpeed } from '../WeatherItemComponent/WeatherWindSpeed.jsx';
import { WeatherWindDirection } from '../WeatherItemComponent/WeatherWindDirection.jsx';
import { getMonthNum } from '../utils/getMonthNum';
import { getDateNum } from '../utils/getDateNum';
import { getDayNum } from '../utils/getDayNum';
import { getYearNum } from '../utils/getYearNum';
import { transformDate } from '../utils/transformDate.js';

Enzyme.configure({ adapter: new Adapter() });

describe('WeatherItem', () => {
  let wrapper;
  const item = WEATHER[0];
  const newDate = new Date(WEATHER[0].date);
  const { day, dateNum, month, year } = transformDate(
    newDate,
    getMonthNum,
    getDateNum,
    getDayNum,
    getYearNum
  );

  beforeEach(() => {
    wrapper = mount(<WeatherItem {...item} />);
  });

  test('shoud render weather item wrapper', () => {
    expect(wrapper.find('.weather__item').length).toBe(1);
  });

  test('shoud render day', () => {
    expect(wrapper.find(WeatherDay).length).toBe(1);
    expect(wrapper.find('.weather__day').length).toBe(1);
    expect(wrapper.find('.weather__day').text()).toBe(day);
  });

  test('shoud render date', () => {
    expect(wrapper.find(WeatherDate).length).toBe(1);
    expect(wrapper.find('.weather__date').length).toBe(1);
    expect(wrapper.find('.weather__date').text()).toBe(dateNum.toString());
  });

  test('shoud render month', () => {
    expect(wrapper.find(WeatherMonth).length).toBe(1);
    expect(wrapper.find('.weather__month').length).toBe(1);
    expect(wrapper.find('.weather__month').text()).toBe(month);
  });

  test('shoud render year', () => {
    expect(wrapper.find(WeatherYear).length).toBe(1);
    expect(wrapper.find('.weather__year').length).toBe(1);
    expect(wrapper.find('.weather__year').text()).toBe(`${year} year`);
  });

  test('shoud render description', () => {
    expect(wrapper.find(WeatherDescription).length).toBe(1);
    expect(wrapper.find('.weather__description').length).toBe(1);
    expect(wrapper.find('.weather__description').prop('src')).toBe(
      item.description
    );
  });

  test('shoud render temperature', () => {
    expect(wrapper.find(WeatherTemperature).length).toBe(1);
    expect(wrapper.find('.weather__temperature-wrapper').length).toBe(1);
    expect(wrapper.find('.weather__min-temperature').length).toBe(1);
    expect(wrapper.find('.weather__min-description').length).toBe(1);
    expect(wrapper.find('.weather__min-value').length).toBe(1);
    expect(wrapper.find('.weather__max-temperature').length).toBe(1);
    expect(wrapper.find('.weather__max-description').length).toBe(1);
    expect(wrapper.find('.weather__max-value').length).toBe(1);
    expect(wrapper.find('.weather__max-description').text()).toBe('max');
    expect(wrapper.find('.weather__min-description').text()).toBe('min');
  });

  test('shoud render pressure', () => {
    expect(wrapper.find(WeatherPressure).length).toBe(1);
    expect(wrapper.find('.weather__atmospheric-pressure').length).toBe(1);
    expect(wrapper.find('.weather__atmospheric-pressure').text()).toBe(
      `Atmospheric pressure: ${item.atmosphericPressure}`
    );
  });

  test('shoud render humidity', () => {
    expect(wrapper.find(WeatherHumidity).length).toBe(1);
    expect(wrapper.find('.weather__humidity').length).toBe(1);
    expect(wrapper.find('.weather__humidity').text()).toBe(
      `Reletive humidity: ${item.relativeHumidity}`
    );
  });

  test('shoud render wind speed', () => {
    expect(wrapper.find(WeatherWindSpeed).length).toBe(1);
    expect(wrapper.find('.weather__wind-speed').length).toBe(1);
    expect(wrapper.find('.weather__wind-speed').text()).toBe(
      `Wind speed: ${item.windSpeed}`
    );
  });

  test('shoud render wind direction', () => {
    expect(wrapper.find(WeatherWindDirection).length).toBe(1);
    expect(wrapper.find('.weather__wind-direction').length).toBe(1);
    expect(wrapper.find('.weather__wind-direction').text()).toBe(
      `Wind direction: ${item.windDirection}`
    );
  });

  test('shoud match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
