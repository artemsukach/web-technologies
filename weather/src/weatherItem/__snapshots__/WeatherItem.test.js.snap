// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem shoud match snapshot 1`] = `"<div class=\\"weather__item\\"><p class=\\"weather__day\\">Monday</p><p class=\\"weather__date\\">28</p><p class=\\"weather__month\\">February</p><p class=\\"weather__year\\">2022 year</p><img class=\\"weather__description\\" src=\\"snow.png\\" alt=\\"weather\\"><div class=\\"weather__temperature-wrapper\\"><div class=\\"weather__min-temperature\\"><p class=\\"weather__min-description\\">min</p><p class=\\"weather__min-value\\">-3<span>°</span></p></div><div class=\\"weather__max-temperature\\"><p class=\\"weather__max-description\\">max</p><p class=\\"weather__max-value\\">4<span>°</span></p></div></div><p class=\\"weather__atmospheric-pressure\\">Atmospheric pressure: 720</p><p class=\\"weather__humidity\\">Reletive humidity: 80</p><p class=\\"weather__wind-speed\\">Wind speed: 8</p><p class=\\"weather__wind-direction\\">Wind direction: South</p></div>"`;
